lib/matplotlib/pyplot.py

```diff
@@ -569,6 +569,14 @@ def draw_if_interactive(*args, **kwargs):
     return _get_backend_mod().draw_if_interactive(*args, **kwargs)
 
 
+@overload
+def show() -> None: ...
+
+
+@overload
+def show(block: bool) -> None: ...
```

Comment:
`pyplot_show` (which is what is used to overwrite the signature by `switch_backend`) has [`block` as keyword-only and optional](https://github.com/matplotlib/matplotlib/blob/main/lib/matplotlib/backend_bases.pyi#L426). Why can't this also be a single overload of the same form?

At least, `block` should be keyword-only as well, shouldn't it?